lib/IDE/CodeCompletion.cpp

```diff
@@ -4014,20 +4013,8 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
         Kind, LookupKind::EnumElement);
     NeedLeadingDot = !HaveDot;
 
-    const DeclContext *FunctionDC = CurrDeclContext;
-    const AbstractFunctionDecl *CurrentFunction = nullptr;
-    while (FunctionDC->isLocalContext()) {
-      if (auto *AFD = dyn_cast<AbstractFunctionDecl>(FunctionDC)) {
-        CurrentFunction = AFD;
-        break;
-      }
-      FunctionDC = FunctionDC->getParent();
-    }
-    if (!CurrentFunction)
-      return;
-
```

Comment:
We don't need to find `AbstractFunctionDecl` or `TopLevelCodeDecl` here because we know this completion is called only for `case` pattern, and `switch expr { case #^COMPLETE^#` doesn't have inner decl context. That means current decl context should contain `switch`.